packages/vue/src/utils/deep-to-raw.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  isProxy,
+  isReactive,
+  isRef,
+  toRaw,
+} from 'vue'
+
+export function deepToRaw<T>(input: T): T {
+  if (Array.isArray(input)) {
+    return input.map(deepToRaw) as T
+  }
+
+  if (isRef(input)
+    || isReactive(input)
+    || isProxy(input)) {
+    return deepToRaw(toRaw(input))
+  }
+
+  if (input !== null && typeof input === 'object') {
+    const clone = {}
+    for (const [key, value] of Object.entries(input)) {
+      (clone as Record<string, unknown>)[key] = deepToRaw(value)
+    }
+    return clone as T
+  }
+  return input
+}
```

Comment:
```suggestion
export function deepToRaw<T>(input: T): T {
  if (Array.isArray(input))
    return input.map(deepToRaw) as T

  if (isRef(input) || isReactive(input) || isProxy(input))
    return deepToRaw(toRaw(input))

  if (input !== null && typeof input === 'object')
    return Object.keys(input).reduce((acc, key) => {
      acc[key as keyof typeof acc] = deepToRaw(input[key])
      return acc
    }, {} as T)

  return input
}
```

from https://github.com/vuejs/core/issues/5303#issuecomment-1543596383

Reply:
I tried mixing the codes from both sides